Readme.md

```diff
@@ -3,57 +3,134 @@
-* [DataHelper.cs](./CS/DXWebApplication23/Models/DataHelper.cs) (VB: [DataHelper.vb](./VB/DXWebApplication23/Models/DataHelper.vb))
-* [Index.cshtml](./CS/DXWebApplication23/Views/Home/Index.cshtml)
-* **[SpreadsheetPartial.cshtml](./CS/DXWebApplication23/Views/Home/SpreadsheetPartial.cshtml)**
-<!-- default file list end -->
-# Spreadsheet - How to save and load documents from a database
-
-## Steps to implement:
-1) Create a helper class that returns the Spreadsheet settings. Make sure that you specify the **Name** and the **CallbackRouteValues** properties:
-```cs
- public static class SpreadsheetSettingsHelper {
+This example demonstrates how to configure the [Spreadsheet extension](https://docs.devexpress.com/AspNetMvc/17113/components/spreadsheet) to save and load documents from a database.
+
+![Connect Spreadsheet to Database](connect-spreadsheet-to-database.png)
+
+## Overview
+
+Follow the steps below to configure the Spreadsheet extension to work with a database:
+
+1. Create a class that models a spreadsheet document. This class should be able to store a document's identifier, format, and content:
+
+    ```cs
+    public class SpreadsheetData {
+        public string DocumentId { get; set; }
+        public DocumentFormat DocumentFormat { get; set; }
+        public byte[] Document { get; set; }
+    }
+    ```
+
+2. Register the database context:
+
+    ```cs
+    public partial class DocumentsEntities : DbContext {
+        public DocumentsEntities(): base("name=DocumentsEntities") {}
+        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
+            throw new UnintentionalCodeFirstException();
+        }
+        public virtual DbSet<Doc> Docs { get; set; }
+    }
+
+    public partial class Doc {
+        public int Id { get; set; }
+        public byte[] DocBytes { get; set; }
+        public string Comments { get; set; }
+    }
+    ```
+
+3. Add connection strings to the **Web.config** file:
+
+    ```xml
+    <connectionStrings>
+        <add name="DocumentsConnectionString" connectionString="Data Source=(local);Initial Catalog = Documents;User Id=sa; Password=dx;Connect Timeout=30" providerName="System.Data.SqlClient" />
+        <add name="DocumentsEntities" connectionString="metadata=res://*/Models.Model1.csdl|res://*/Models.Model1.ssdl|res://*/Models.Model1.msl;provider=System.Data.SqlClient;provider connection string=&quot;Data Source=(localdb)\MSSQLLocalDB;AttachDbFilename=|DataDirectory|Documents.mdf;Integrated Security=True;connect timeout=30;MultipleActiveResultSets=True;App=EntityFramework&quot;" providerName="System.Data.EntityClient" />
+    </connectionStrings>
+    ```
+
+4. Create a helper class that works with the database. The class should be able to get a document from a database as an array of bytes and save a byte array back to the database:
```

Comment:
> The class should be able to get a document from a database as an array of bytes and save **a** byte array **back** to the database:

**A potential issue**. At first, everything looks the way you describe the abstract functionality of your helper class. The class should be able to load a document as an array of bytes and save a byte array to the database.

However, you said "back" in "save a byte array **back** to the database". If we save something "back", it implies that we speak of the same array of bytes, or the same document. In this case, a definite article is in order. Or getting rid of "back" as an indication that implies definiteness in this particular usage scenario.

```suggestion
4. Create a helper class that works with the database. The class should be able to load a document from a database as an array of bytes and save the byte array back to the database:
```

OR

```suggestion
4. Create a helper class that works with the database. The class should be able to load a document from a database as an array of bytes and save a document as a byte array to the database:
```

Again, I'm OK with either variant, however, you may come up with something better.

P.S. Why "get" a document rather than "load"? "Load" is the opposed operation to "Save" after all.